src/algebraic_topology/dold_kan/compatibility.lean

```diff
@@ -176,6 +176,119 @@ end
 
 lemma equivalence_functor : (equivalence hF hG).functor = F ⋙ eB.inverse := rfl
 
+omit hG hF
+
+/-- The isomorphism `eB.functor ⋙ e'.inverse ⋙ e'.functor ≅ eB.functor` deduced
+from the counit isomorphism of `e'`. -/
+@[simps hom_app]
+def τ₀ : eB.functor ⋙ e'.inverse ⋙ e'.functor ≅ eB.functor :=
+calc eB.functor ⋙ e'.inverse ⋙ e'.functor
+  ≅ eB.functor ⋙ 𝟭 _ : iso_whisker_left _ e'.counit_iso
+... ≅ eB.functor : functor.right_unitor _
+
+include hF hG
+
+/-- The isomorphism `eB.functor ⋙ e'.inverse ⋙ e'.functor ≅ eB.functor` deduced
+the isomorphisms `hF : eA.functor ⋙ e'.functor ≅ F`,
+`hG : eB.functor ⋙ e'.inverse ≅ G ⋙ eA.functor` and the datum of
+an isomorphism `η : G ⋙ F ≅ eB.functor`. -/
+@[simps hom_app]
+def τ₁ (η : G ⋙ F ≅ eB.functor) :
+  eB.functor ⋙ e'.inverse ⋙ e'.functor ≅ eB.functor :=
+calc eB.functor ⋙ e'.inverse ⋙ e'.functor
+  ≅ (eB.functor ⋙ e'.inverse) ⋙ e'.functor : iso.refl _
+... ≅ (G ⋙ eA.functor) ⋙ e'.functor : iso_whisker_right hG _
+... ≅ G ⋙ (eA.functor ⋙ e'.functor) : by refl
+... ≅ G ⋙ F : iso_whisker_left _ hF
+... ≅ eB.functor : η
+
+variables (η : G ⋙ F ≅ eB.functor) (hη : τ₀ = τ₁ hF hG η)
+
+omit hF hG
+include η
+
+/-- The counit isomorphism of `equivalence`. -/
+@[simps]
+def equivalence_counit_iso : G ⋙ (F ⋙ eB.inverse) ≅ 𝟭 B :=
+calc G ⋙ (F ⋙ eB.inverse) ≅ (G ⋙ F) ⋙ eB.inverse : iso.refl _
+... ≅ eB.functor ⋙ eB.inverse : iso_whisker_right η _
+... ≅ 𝟭 B : eB.unit_iso.symm
+
+variables {η hF hG}
+include hη
+
+lemma equivalence_counit_iso_eq :
+  (equivalence hF hG).counit_iso = equivalence_counit_iso η :=
+begin
+  ext1, apply nat_trans.ext, ext Y,
+  dsimp [equivalence, equivalence_counit_iso, is_equivalence.of_equivalence],
+  simp only [equivalence₂_counit_iso_eq eB hF],
+  erw [nat_trans.id_app, nat_trans.id_app],
+  dsimp [equivalence₂, equivalence₁],
+  simp only [assoc, comp_id, F.map_id, id_comp,
+    equivalence₂_counit_iso_hom_app, ← eB.inverse.map_comp_assoc,
+    ← τ₀_hom_app, hη, τ₁_hom_app],
+  erw hF.inv.naturality_assoc,
+  congr' 2,
+  dsimp,
+  simp only [assoc, ← e'.functor.map_comp_assoc, eA.functor.map_comp,
+    equivalence.fun_inv_map, iso.inv_hom_id_app_assoc, hG.inv_hom_id_app],
+  dsimp,
+  rw [comp_id, eA.functor_unit_iso_comp, e'.functor.map_id, id_comp, hF.inv_hom_id_app_assoc],
+end
```

Comment:
Wow! Well done, it doesn't even take forever to compile. Could you imagine a tactic doing this kind of argument in Lean 4?

Reply:
It was more painful when I initially attempted to prove these compatibilities directly on the Dold-Kan functors, but it become more smooth when I isolated the "general" statement involving abstract functors and isomorphisms. I do not know how to even think of a `category_coherence` which would prove all diagrams involving adjunctions commute.